developer_guides/virtualisation/virtualisation.rst

```diff
@@ -0,0 +1,349 @@
+.. _virtualisation:
+
+SOF VirtIO design
+#################
+
+.. contents::
+
+1. General information
+**********************
+
+VirtIO_ is a platform-agnostic virtualisation standard, managed by OASIS_. It
+defines a design and a protocol for implementing device IO in VMs.
+
+There are different machine-level virtualisation designs, implementing different
+software configurations. Typically they will have one instance of a general
+purpose OS, running either directly on the hardware, or at least with direct
+unrestricted access to the hardware. This instance is usually called a "host."
+Within that host OS, there is usually one or more user-space processes,
+emulating further independent virtual machines, running their own complete
+instances of general-purpose OSes. Those instances are called "guests."
+
+When Linux is running in a VM, multiple drivers are already available, using
+virtio: networking, storage, etc. Those drivers use the virtio and virtqueue
+APIs to communicate with the host or, potentially, with other guests.
+
+The host part depends on the used virtualisation solution. Some of them
+implement the VirtIO API themselves, then using their method of choice to talk
+to the host they are running on. A typical Linux virtualisation implementation
+uses QEMU + KVM to run guest instances. In this case there are two ways to
+implement a VirtIO host: in the user space and in the kernel. If implemented in
+the user space, it is managed by QEMU. In the kernel space VirtIO host drivers
+use the Linux vhost API.
+
+It is also worth to note, that VirtIO guest drivers reside in respective driver
+subdirectories in the Linux kernel, e.g. drivers/net, drivers/block etc. On the
+contrary, all vhost drivers are located under drivers/vhost. That directory also
+contains vhost.c and vhost.h which are used by all vhost drivers.
+
+This document describes the Linux driver architecture to virtualise SOF DSP
```

Comment:
This should be at start of the document. all the way down to....

Reply:
@lrgirdwo Should I then throw away the explanation of VirtIO in Linux? It would be wrong to have that after this text. Or we can add a "Preliminary information" subsection?

Reply:
it can be kept, the main point of this document is to describe SOF virtualisation and not virto, so SOF stuff would be the first paragraph.